dsgrid/config/dataset_schema_handler_one_table.py

```diff
@@ -0,0 +1,82 @@
+from pathlib import Path
+
+from dsgrid.config.dataset_config import (
+    DatasetConfig,
+    check_load_data_filename,
+    check_load_data_lookup_filename,
+)
+from dsgrid.utils.spark import read_dataframe, get_unique_values
+from dsgrid.utils.timing import timer_stats_collector, Timer
+from dsgrid.config.dataset_schema_handler_base import DatasetSchemaHandlerBase
+from dsgrid.dimension.base_models import DimensionType
+
+
+class OneTableDatasetSchemaHandler(DatasetSchemaHandlerBase):
+    """ define interface/required behaviors for ONE_TABLE dataset schema """
+
+    def __init__(self, config):
+        self._config = config
+
+    def check_consistency(self):
+        path = Path(self._config.model.path)
+        load_data_df = read_dataframe(check_load_data_filename(path), cache=True)
+        load_data_df = self._config.add_trivial_dimensions(load_data_df)
+        with Timer(timer_stats_collector, "check_one_table_data_consistency"):
+            self._check_one_table_data_consistency(self._config, load_data_df)
+        with Timer(timer_stats_collector, "check_dataset_time_consistency"):
+            self._check_dataset_time_consistency(self._config, load_data_df)
+
+    def _check_one_table_data_consistency(self, config: DatasetConfig, load_data):
+        dimension_types = []
+        metric_cols = []
+
+        time_dim = config.get_dimension(DimensionType.TIME)
+        time_columns = set(time_dim.get_timestamp_load_data_columns())
+
+        for col in load_data.columns:
+            if col in time_columns:
+                continue
+            try:
+                dimension_types.append(DimensionType(col))
+            except ValueError:
+                metric_cols.append(col)
```

Comment:
Can you check for load value columns explicitly? Also, are they guaranteed to be of type metric?

Reply:
No, we just assume all that have ValueError are of metric type. They are collected and then compared to the metric dimension records. https://github.com/dsgrid/dsgrid/blob/e5f4080c6432f42d2782bb345451a168bbea5e5b/dsgrid/config/dataset_schema_handler_one_table.py#L66-L75

Reply:
Oh, and I guess you are assuming that the MetricType dimension is the one that is pivoted out. I thought we might have/were planning to make that configurable? (E.g., list in the dataset config which dimension type is pivoted out to form the value columns.)

Reply:
As discussed this afternoon, the schema should specify which dimension type is pivoted, and somewhere the dataset (speaking loosely) should be able to specify what column names it is expecting by combining the dataset config with the dimension records.

Reply:
Got it, I'll use `load_data_column_dimension`, which the standard schema is already using